python/packages/autogen-ext/tests/tools/test_mcp_tools.py

```diff
@@ -595,3 +595,58 @@ async def test_lazy_init_and_finalize_cleanup() -> None:
     assert workbench._actor._active is True  # type: ignore[reportPrivateUsage]
 
     del workbench
+
+
+def test_mcp_tool_adapter_normalize_payload(sample_tool: Tool, sample_server_params: StdioServerParams) -> None:
+    """Test the _normalize_payload_to_content_list method of McpToolAdapter."""
+    adapter = StdioMcpToolAdapter(server_params=sample_server_params, tool=sample_tool)
+
+    # Case 1: Payload is already a list of valid content items
+    valid_content_list = [
+        TextContent(text="hello", type="text"),
+        ImageContent(data="base64data", mimeType="image/png", type="image"),
+        EmbeddedResource(
+            type="resource",
+            resource=TextResourceContents(text="embedded text", uri=AnyUrl(url="http://example.com/resource")),
+        ),
+    ]
+    assert adapter._normalize_payload_to_content_list(valid_content_list) == valid_content_list
```

Comment:
PR is looking good @peterj 
There are a few pyright/mypy issues that need addressing.
you can run `poe pyright` and `poe mypy` in the `packages/autogen-ext` file to test those locally. 

Let me know if there are any issues and I can help with finalizing.